Source/Core/UICommon/ResourcePack/ResourcePack.cpp

```diff
@@ -139,14 +141,44 @@ bool ResourcePack::Install(const std::string& path)
     return false;
   }
 
+  // TODO would strm_buffered interface give better perf?
   auto file = unzOpen(m_path.c_str());
+  if (file == nullptr)
+  {
+    m_valid = false;
+    m_error = "Failed to open resource pack";
+    return false;
+  }
   Common::ScopeGuard file_guard{[&] { unzClose(file); }};
 
-  for (const auto& texture : m_textures)
+  if (unzGoToFirstFile(file) != MZ_OK)
+    return false;
+  do
   {
-    bool provided_by_other_pack = false;
+    std::string texture_zip_path(UINT16_MAX + 1, '\0');
+    unz_file_info64 texture_info{};
+    if (unzGetCurrentFileInfo64(file, &texture_info, texture_zip_path.data(), UINT16_MAX, nullptr,
+                                0, nullptr, 0) != MZ_OK)
+    {
+      return false;
+    }
+    TruncateToCString(&texture_zip_path);
+
+    const std::string texture_zip_path_prefix = "textures/";
+    if (!texture_zip_path.starts_with(texture_zip_path_prefix))
+      continue;
+    const std::string texture_name = texture_zip_path.substr(texture_zip_path_prefix.size());
```

Comment:
Micro-optimizing a bit here but do we need to create a new string?  Maybe:

```
std::string_view t_zip_path = texture_zip_path;
const std::string_view texture_name = t_zip_path.substr(texture_zip_path_prefix.size());
```

would be faster?

Reply:
yea good point, i'm just not very familiar with string_view

Reply:
actually i dont think it helps? since i need `c_str` from it and string_view i think by definition doesn't provide a way to get the view with a null terminator

Reply:
Ah dang.  Wasn't even thinking of null termination being an issue.  Sorry about that!